tests/sentry/sentry_metrics/querying/data_v2/test_api.py

```diff
@@ -1391,9 +1401,13 @@ def test_query_with_formula_and_filter(self):
         assert data[0][1]["by"] == {"platform": "android", "transaction": "/hello"}
         assert data[0][1]["series"] == [None, 1.0, 1.0]
         assert data[0][1]["totals"] == 2.0
-        assert data[0][2]["by"] == {"platform": "windows", "transaction": "/world"}
-        assert data[0][2]["series"] == [None, 0.0, 0.0]
-        assert data[0][2]["totals"] == 0.0
+
+        # TODO: after snuba pr #5676 is merged, change to assert len(data[0]) == 2
+        assert len(data[0]) in {2, 3}
+        if len(data[0]) == 3:
+            assert data[0][2]["by"] == {"platform": "windows", "transaction": "/world"}
+            assert data[0][2]["series"] == [None, 0.0, 0.0]
+            assert data[0][2]["totals"] == 0.0
```

Comment:
Wouldn't it be better to just put the `xfail` on this test and like in the other tests?

Adding the assert `len(data[0]) == 2` before checking the data of each timeseries and letting it fail.

Reply:
yeah thats probably better, will change